libnmstate/nm/sriov.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import Ethernet
+from libnmstate.schema import SRIOV
+
+
+def create_setting(iface_state, base_con_profile):
+    ethernet_setting = iface_state.get(Ethernet.CONFIG_SUBTREE)
+    if not ethernet_setting:
+        return None
+
+    setting = iface_state.get(SRIOV.CONFIG_SUBTREE)
+    sriov_setting = _get_current_sriov_setting(base_con_profile)
+    if not setting:
+        return sriov_setting
+
+    if not sriov_setting:
+        sriov_setting = nmclient.NM.SettingSriov.new()
+
+    if setting.total_vfs:
+        sriov_setting.props.total_vfs = setting.total_vfs
+
+    return sriov_setting
+
+
+def _get_current_sriov_setting(base_con_profile):
+    sriov_setting = None
+    if base_con_profile:
+        sriov_setting = base_con_profile.get_setting_by_name(
+            nmclient.NM.SETTING_SRIOV_SETTING_NAME
+        )
+        if sriov_setting:
+            sriov_setting = sriov_setting.duplicate()
+    return sriov_setting
+
+
+def get_info(device):
+    """
+    Provide the current active SR-IOV values for a device
+    """
+    info = {}
+
+    connection = nm_connection.ConnectionProfile()
```

Comment:
Can we get the SR-IOV settings from the device/interface (IOW the actual values used by the kernel, or is it only available currently from the configuration?)

Maybe @thom311 can help, it might need a RFE from NM.

Reply:
AFAIK, currently libnm only allows us to get the settings from the configuration. If I am wrong, please let me know @thom311. Could be that a problem?

Reply:
no, you cannot... and RFE would be good.

Reply:
@ffmancera please file an RFE in Bugzilla for NM, another Bug for Nmstate that depends on the NM RFE. Please also mention it in the code.

Reply:
Will do. Thanks!

Reply:
Done!